game.py

```diff
@@ -411,22 +424,163 @@ def check_ghost_collisions(self):
                 self.reset_positions()
 
     def move_ghost(self, ghost):
-        if ghost.alive == False:
-            # if the ghost is "dead" then we should teleport it back to
-            # its starting location in the house. 
-            ghost.pos[0] = ghost.start_x
-            ghost.pos[1] = ghost.start_y
-            ghost.vulnerable = 0
-            ghost.alive = True
-            ghost.in_house = True
+            # Ghost movement reference: https://gameinternals.com/understanding-pac-man-ghost-behavior
+            # Known deficiencies:
+            #   Pinky and Inky do not chase properly
+            #   
+
+        def remove_backwards(dirs):
+            '''Given a list of directions to move remove the option to reverse direction'''
+            if len(dirs) > 1:
+                if ghost.previous_dir == 'w' and 's' in dirs:
+                    dirs.remove('s')
+                elif ghost.previous_dir == 's' and 'w' in dirs:
+                    dirs.remove('w')
+                elif ghost.previous_dir == 'a' and 'd' in dirs:
+                    dirs.remove('d')
+                elif ghost.previous_dir == 'd' and 'a' in dirs:
+                    dirs.remove('a')
+            
+            return dirs
+        
+        def dir_to_target():
+            '''Find the legal moves that are closest to the current target.'''
+            min_dist = self.map_distance(ghost.pos, ghost.target_pos) + 100
```

Comment:
Why `+100`? Is this just to make sure this is a large number? If so why not use `math.inf`?

Reply:
You're correct, that should absolutely be `math.inf`